### PR TITLE
Modify "feature request" template for more natural user experience

### PR DESCRIPTION
**Pull request by qianyi-sun**
The structure of current "feature request" issue template is "Proposal" then "Motivation", which is easier for us to read as devs because when we check it we can always see the solution that user would like first. 

But from user's perspective, it might be more logically natural for general writing if motivation/context goes first, and followed by the possible solution. 

Still not sure if we should do this since either way has its pros/cons, sall comments are welcomed!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 1]
```diff
@@ -3,6 +3,13 @@ description: Suggest an idea for this project not related to bugs
 title: "[Feature request] Feature request title"
 labels: ["enhancement"]
 body:
+  - type: textarea
+    id: motivation
+    attributes:
+      label: Motivation
+      description: |
+        Is your feature request related to a problem? Please describe. 
+        Give clear and concise description of what the problem is. Ex. I am always frustrated when (...).
```
Comment (Gamenot) [inline .github/ISSUE_TEMPLATE/feature_request.yml:12]: If we are moving it up here we must force the user to justify the change.
```suggestion
  - type: textarea
    id: motivation
    attributes:
      label: Motivation
      description: |
        Is your feature request related to a problem? Please describe. 
        Give clear and concise description of what the problem is. Ex. I am always frustrated when (...).
    validations:
      required: true
```

**Review by saulfield — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 1 above)
Comment (Adaickalavan) [reply .github/ISSUE_TEMPLATE/feature_request.yml:12]: Requesting users to describe the issue will allow us to evaluate the proposed changes and also help us formulate our own proposals.

Comment (Gamenot) [reply .github/ISSUE_TEMPLATE/feature_request.yml:12]: It is difficult to prioritize or offer alternatives if we do not know why the user is making the request. My opinion is that the motivation statement is more important than the proposal.

**Review by Adaickalavan — APPROVED** [review]

**Comment by Gamenot** [comment]
@qianyi-sun Please merge this when you have the chance.

**Pull request merged**